### PR TITLE
Update rocksdb to 0.19 (#7357)

### PR DESCRIPTION
**Pull request by mina86**
This brings a few changes which allow our code to be simplified:

* Iterators now return Result<Item, Error> rather than forcing users
  to check for errors with separate status method call.  We can remove
  our wrapping code;

* rocksdb::PrefixRange allows setting a prefix range which is what we
  used to do ourselves.  This lets us get rid of next_prefix function;

* property_value method et al. accept CStr arguments and properties
  defined in rocksdb::properties are now defined as CStrs.  This is an
  optimisation as previously each call to property_int_value_cf
  resulted in str→CStr conversion which requires memory allocation.

Notably, this forces upgrade of tikv-jemallocator dependency from
0.4.3 to 0.5.0 and actix-http from 3.0.4 to 3.2.2.  This latter is due
to conflicts with zstd-sys crate which rocksdb and actix-http both
(transitively) depend on.

Fixes: https://github.com/near/nearcore/issues/7718

### CODE REVIEW

**Comment by mina86** [comment]
This seems to be fixing some NayDuck failures: https://nayduck.near.org/#/run/2701

File: core/store/src/db/rocksdb.rs [hunk 1]
```diff
@@ -202,52 +202,51 @@ impl RocksDB {
...
-            status.err().map(into_other).map(Result::Err)
-        }
+        Some(self.0.next()?.map_err(into_other))
     }
 }
 
 impl<'a> std::iter::FusedIterator for RocksDBIterator<'a> {}
 
+impl RocksDB {
+    fn get_cf_key_range(
```
Comment (mm-near) [inline core/store/src/db/rocksdb.rs:231]: add comment

File: deny.toml [hunk 2]
```diff
@@ -100,4 +100,7 @@ skip = [
 
     # opentelemetry-otlp depends on an old version of tonic which depends on an old version of tokio-util
     { name = "tokio-util", version = "=0.6.10" },
+
+    # redis we’re using uses ancient sha
+    { name = "sha1", version = "=0.6.1" },
```
Comment (mm-near) [inline deny.toml:105]: is this related to this PR?

File: core/store/src/db/rocksdb.rs [hunk 3]
```diff
@@ -298,13 +297,11 @@ impl Database for RocksDB {
                 }
                 DBOp::DeleteAll { col } => {
                     let cf_handle = self.cf_handle(col)?;
-                    let opt_first = self.db.iterator_cf(cf_handle, IteratorMode::Start).next();
-                    let opt_last = self.db.iterator_cf(cf_handle, IteratorMode::End).next();
-                    assert_eq!(opt_first.is_some(), opt_last.is_some());
-                    if let (Some((min_key, _)), Some((max_key, _))) = (opt_first, opt_last) {
-                        batch.delete_range_cf(cf_handle, &min_key, &max_key);
+                    let range = self.get_cf_key_range(cf_handle).map_err(into_other)?;
```
Comment (mm-near) [inline core/store/src/db/rocksdb.rs:300]: so if this returns 'Ok(None)' - we're not doing any deletions?

shoudl we print some warning / error then? Under what conditions does it happen?

**Review by mm-near — APPROVED** [review]

File: deny.toml (hunk 2 above)
Comment (mina86) [reply deny.toml:105]: Yes, though I’m not completely sure why to be honest.  `cargo build` decided that actix-http needs to be updated to 3.2 which changed its dependency from `sha-1` to `sha1 0.10.4` so now we depend on two different versions of `sha1`.  What I don’t understand is why Cargo decided to update actix-http.

File: core/store/src/db/rocksdb.rs (hunk 3 above)
Comment (mina86) [reply core/store/src/db/rocksdb.rs:300]: `None` means that the column is empty so it’s normal to not perform any deletions in that case.

File: deny.toml (hunk 2 above)
Comment (mina86) [reply deny.toml:105]: I’ve tracked it a little bit more.  What happens is conflict with zstd-sys.  actix-http and rocksdb both depend on it.  Updating RocksDB changes version requirement for that crate to ones which no longer work for actix-http 3.0 so it too needs to be updated.
